src/resonate.ts

```diff
@@ -233,7 +233,25 @@ export class Resonate {
       strict: false,
     });
 
-    return this.handle(promiseHandler);
+    return this.handle(id, promiseHandler);
+  }
+
+  /**
+   * Get a promise and return a value
+   */
+  public async get<T = any>(id: string): Promise<ResonateHandle<T>> {
+    let promiseHandler: PromiseHandler;
+
+    try {
+      promiseHandler = await this.inner.get({
+        kind: "readPromise",
+        id,
+      });
+    } catch (err) {
+      throw new Error(`Promise '${id}' not found`);
```

Comment:
The exceptions and observations work will formalize this error, for now this will do